lib/semitran.gd

```diff
@@ -17,8 +17,6 @@ DeclareSynonym("IsTransformationMonoid", IsMonoid and
 IsTransformationCollection);
 
 DeclareProperty("IsFullTransformationSemigroup", IsSemigroup);
-InstallTrueMethod(IsSemigroup, IsFullTransformationSemigroup);
```

Comment:
Why is this unnecessary?

Reply:
Perhaps I misunderstand how things work.

Since `IsFullTransformationSemigroup` is only declared for `IsSemigroup`, that means anything that could know it satisfies `IsFullTransformationSemigroup` already knows that it is a semigroup.

Is perhaps the idea that: this `InstallTrueMethod` is useful because perhaps someone in a package could (for example) do `DeclareProperty("IsFullTransformationSemigroup", IsMagma);` and then install methods for `IsMagma` (which would have to somehow test the magma for associativity, but perhaps not directly, and this could therefore mean that `IsFullTransformationSemigroup` gets set but not `IsSemigroup`)?

But actually it seems that any magma of transformations already knows that it is associative, and therefore a semigroup:
```
gap> S := Magma(GeneratorsOfMagma(FullTransformationSemigroup(3)));;
gap> HasIsSemigroup(S);
true
```
I can't think of any use for this true method, but perhaps I am not imaginative enough 🙂 

I notice that there are similar true methods for natural symmetric and alternating groups (e.g. in `lib/gpprmsya.gd`):
```
InstallTrueMethod( IsPermGroup, IsNaturalSymmetricGroup );
```
But things seem to work differently for groups:
```
gap> G := Semigroup(GeneratorsOfMagma(SymmetricGroup(4)));;
gap> IsGroup(G);
false
gap> SetIsNaturalSymmetricGroup(G, true);
gap> IsGroup(G);
true
```

Reply:
There is no automatic implication and there cannot be, as indeed GAP cannot predict the future and so can't know whether that property might in the future also be applicable to, say, algebras or strings or graphs or... Sure, as a human you know this is unlikely because of the name, but GAP does not known this (and anyway.

Somewhat related: issue #2336

Reply:
> (and anyway.

Were you going to add something?

Anyway, I'll be happy to remove this commit. I'll read #2336 too.